content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,325 @@
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+        - name: "foo3"
+          value: "bar3"
+```
+
+But the next one would be accepted:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+        - name: "foo2"
+          value: "bar2"
+```
+
+### allAreUsed
+
+`allAreUsed` the container cannot use all listed environment variables at once. Checks if all of the `environmentVariables` are in the Pod/Workload resource:
+
+For example, given the following configuration:
+
+```yaml
+settings:
+  rules:
+    - reject: allAreUsed
+      environmentVariables:
+        - name: "foo"
+          value: "bar"
+        - name: "foo2"
+          value: "bar2"
+        - name: "foo3"
+          value: "bar3"
+```
+
+The following Pod would be rejected:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+        - name: "foo2"
+          value: "bar2"
+        - name: "foo3"
+          value: "bar3"
+```
+
+But the next one would be accepted:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+        - name: "foo3"
+          value: "bar3"
+```
+
+
+
+### notAllAreUsed
+
+`notAllAreUsed` the container can use all listed environment variables at once, but only one of them. Checks if all of the `environmentVariables` are not in the Pod/Workload resource
+
+For example, given the following configuration:
+
+```yaml
+settings:
+  rules:
+    - reject: notAllAreUsed
+      environmentVariables:
+        - name: "foo"
+          value: "bar"
+        - name: "foo2"
+          value: "bar2"
+```
+
+The following Pod would be rejected:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  spec:
+    containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+```
+
+But the next one would be accepted:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      ports:
+      - containerPort: 80
+      env:
+        - name: "foo"
+          value: "bar"
+        - name: "foo2"
+          value: "bar2"
+```
+
+## Using multiple rules
+
+The user is allowed to define multiple rules in the policy settings. The resource must be approved by all of them in order to be accepted into the cluster.
```

Comment:
Can you add an example of that? This can be particularly tricky to understand

Reply:
Done